### DISCUSSION TITLE
How do you insert an auth.users in a seed file?

### DISCUSSION THREAD

**Discussion by KrisCoulson** [supabase/supabase#1323] [Questions] [answered]
I have been using `supabase cli` to run supabase locally. I am wondering what is the best way insert an authenticated user during seeding my database. Being able to seed a user locally would definitely help contributors get onboarded much faster.

Currently in my seed file I am able to connect my `public."User"` table to `auth.users` table 

```
  await db.$executeRaw(
    'create or replace function public.handle_new_user() returns trigger language plpgsql security definer as $function$ begin insert into public."User" (uuid) values (new.id); return new; end; $function$'
  )

  await db.$executeRaw(
    'create trigger on_auth_user_created after insert on auth.users for each row execute procedure public.handle_new_user();'
  ) 
 ```
 
 After doing this I would love to create a user. Simply inserting a user I don't think works because the password is not encrypted or I might be missing potential needed fields to login.

**Comment by silentworks** [comment, marked as answer]
Is this for new users to your system or are you migrating from another database to Supabase? 

if new users you are probably best off doing a loop with all the users and calling the `await supabase.auth.api.inviteUserByEmail('email_address_here')` with their email address and provide a page for them to complete the process.

If they are existing users then turn off email confirmation and do a loop with all the users again while calling `await supabase.auth.signUp({ email, password })` and let Supabase deal with the encryption for you. 

Otherwise, if you want to do a straight DB insertion you will need to find out what encryption function is being used for the passwords and how you can call it inside of your SQL migration.

**Reply by KrisCoulson** [reply]
I just went and add the supabase client to my api side and ran `supabase.auth.signUp({ email, password })` and that seemed to do the trick. Dunno why I didn't try that in the first place! Thanks for your help @silentworks

**Comment by KrisCoulson** [comment]
@silentworks yeah this is a fresh DB, essentially anytime I reset and clear my database I want to re-insert the same user. I was attempting to do it via SQL insert but as you mention I am not exactly sure how to encrypt the password
